### PR TITLE
Proposal for Fixing issue #672

### PR DESCRIPTION
**Pull request by pborreli**
Just tested with my own test case : self hosted gitlab

### CODE REVIEW

File: src/GitHub_Updater/API/GitLab_API.php [hunk 1]
```diff
@@ -73,7 +73,8 @@ protected function set_default_credentials() {
 			$set_credentials                            = true;
 		}
 		if ( ( empty( static::$options['gitlab_enterprise_token'] ) &&
-		       ! empty( $this->type->enterprise ) ) ||
+		       ! empty( $this->type->enterprise ) &&
```
Comment (afragen) [inline src/GitHub_Updater/API/GitLab_API.php]: This line would be `in_array( ‘gitlab_enterprise’, $running_servers, true ) &&`

Comment (afragen) [reply src/GitHub_Updater/API/GitLab_API.php]: Assuming the change below.

File: src/GitHub_Updater/Settings.php [hunk 2]
```diff
@@ -134,6 +134,8 @@ private function settings_sub_tabs() {
 		}
 
 		foreach ( $gits as $git ) {
+			$parts = explode('-', $git);
+			$git = $parts[0];
```
Comment (afragen) [inline src/GitHub_Updater/Settings.php]: Do we really need this? Shouldn’t the “GitLab” tab already display?

File: src/GitHub_Updater/Base.php [hunk 3]
```diff
@@ -1270,6 +1270,9 @@ public function get_running_git_servers() {
 			if ( ! empty( $e->enterprise ) && false !== stripos( $e->type, 'bitbucket' ) ) {
 				return 'bbserver';
 			}
+			if (! empty( $e->enterprise ) && 'gitlab.com' !== $e->enterprise ) {
+				return 'gitlab-selfhosted';
+			}
```
Comment (afragen) [inline src/GitHub_Updater/Base.php]: Let’s stick to the same logic for the conditional using `stripos( $e->type, ‘gitlab’ )` and return `gitlab_enterprise` as above.

**Comment by pborreli** [comment]
not convinced at all by this solution, made more test and it doesn't work when more there is more than 1 repo on gitlab hosted

**Pull request closed without merging**